pkg/controllers/networking/pod_controller.go

```diff
@@ -120,7 +148,7 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result
 
 		if feature.VMIPRetainEnabled() {
 			// TODO: use APIReader to get VM/VMI object, because watch v1.VirtualMachine and v1.VirtualMachineInstance will always get errors
-			if isVMPod, vmName, _, err := strategy.OwnByVirtualMachine(ctx, pod, r.APIReader); isVMPod {
+			if isVMPod, vmName, _, err := strategy.OwnByVirtualMachine(ctx, ownedObj, r.APIReader); isVMPod {
```

Comment:
`strategy.OwnByVirtualMachine` should not be used to IPInstance, because IPInstance will be owned by VM but Pod will be owned by VMI (and VMI is owned by VM).

Reply:
Refactored `strategy.OwnByVirtualMachine` a bit to handle directly and indirectly owned relationship.